sparse_strips/vello_common/src/encode.rs

```diff
@@ -394,25 +422,28 @@ fn encode_stops(stops: &[ColorStop], start: f32, end: f32, pad: bool) -> Vec<Gra
         // We handle padding by inserting dummy stops in the beginning and end with a very big
         // range.
         let left_range = iter::once({
-            let first_stop = stops.first().unwrap();
+            let first_stop = interpolated_stops.first().unwrap();
             let mut encoded_range = create_range(first_stop, first_stop);
-            encoded_range.x0 = f32::MIN;
+            encoded_range[0].x0 = f32::MIN;
 
             encoded_range
         });
 
         let right_range = iter::once({
-            let last_stop = stops.last().unwrap();
+            let last_stop = interpolated_stops.last().unwrap();
 
             let mut encoded_range = create_range(last_stop, last_stop);
-            encoded_range.x1 = f32::MAX;
+            encoded_range[0].x1 = f32::MAX;
 
             encoded_range
         });
 
-        left_range.chain(stop_ranges.chain(right_range)).collect()
+        left_range
+            .chain(stop_ranges.chain(right_range))
+            .flatten()
+            .collect()
     } else {
-        stop_ranges.collect()
+        stop_ranges.flatten().collect()
```

Comment:
Using a `GradientRange` from `create_range` directly:

```suggestion
        // We handle padding by inserting dummy stops in the beginning and end with a very big
        // range.
        let left_range = iter::once({
            let first_stop = interpolated_stops.first().unwrap();
            let mut encoded_range = create_range(first_stop, first_stop);
            encoded_range.x0 = f32::MIN;

            encoded_range
        });

        let right_range = iter::once({
            let last_stop = interpolated_stops.last().unwrap();

            let mut encoded_range = create_range(last_stop, last_stop);
            encoded_range.x1 = f32::MAX;

            encoded_range
        });

        left_range.chain(stop_ranges.chain(right_range)).collect()
    } else {
        stop_ranges.collect()
```

Reply:
Good catch!